clang/lib/CConv/ConstraintBuilder.cpp

```diff
@@ -177,6 +160,12 @@ class FunctionVisitor : public RecursiveASTVisitor<FunctionVisitor> {
       FuncName = DD->getNameAsString();
     }
 
+    // Collect type parameters for this function call that are
+    // consistently instantiated as single type in this function call.
+    std::set<unsigned int> consistentTypeParams;
+    if (TFD != nullptr)
+      TVInfo.getConsistentTypeParams(E, consistentTypeParams);
+
```

Comment:
Why is this code here, and not below, where the code used to be? I would think you need to know the function you are calling to be able to do this, and at the position below you know it (you've gotten the `FVConstraint` by then).

Reply:
The code as written maps each class expression to a set of type variables without differentiating between multiple FunctionVariables in a single call. In other words, function pointers aren't handled.

Are generic function pointers allowed in checked c?

Reply:
Not sure. Try it out?